### DISCUSSION TITLE
OpenVPN for Android with tasker

### DISCUSSION THREAD

**Discussion by update-freak** [schwabe/ics-openvpn#1511] [Q&A]
Hello everyone,

how is it possible to connect/disconnect to VPN with Tasker?
I found this information https://github.com/schwabe/ics-openvpn#controlling-from-external-apps= and tried the following:

Send intent with
action: de.blinkt.openvpn.api.ConnectVPN
extra: de.blinkt.openvpn.api.profileName
target: activity

I also tried
Extra: de.blinkt.openvpn.api.MeinNetzwerk
and
de.blinkt.openvpn.api.profileName:MeinNetzwerk
and
{de.blinkt.openvpn.api.profileName:MeinNetzwerk}

Is there anything i missed?

**Comment by PhilC813** [comment]
It's an old thread but in case anyone else is looking for the answer, here's what worked for me:
> **Action**: _android.intent.action.MAIN_ (but also seems to work if left blank)
> **Extra**: _de.blinkt.openvpn.api.profileName:[name of the profile]_
> **Package**: _de.blinkt.openvpn_
> **Class** (to connect): _de.blinkt.openvpn.api.ConnectVPN_
> **Class** (to disconnect):
_de.blinkt.openvpn.api.DisconnectVPN_
> **Target**: _Activity_

Thanks to this other [thread](https://github.com/schwabe/ics-openvpn/discussions/1532) for the guidance!

**Reply by DustinSwick** [reply]
While this may be an old thread, I just wanted to say Thank You for sharing what worked for you, as this worked perfectly for me as well (as of December 2025)!
